tests/kokkos-based/swap_elements_rank1_kokkos.cpp

```diff
@@ -0,0 +1,70 @@
+
+#include "gtest_fixtures.hpp"
+#include "helpers.hpp"
+
+namespace{
+
+template<class x_t, class y_t>
+void kokkos_blas_swap_test_rank1_impl(x_t x, y_t y)
+{
+  namespace stdla = std::experimental::linalg;
+
+  using value_type = typename x_t::value_type;
+  const std::size_t extent = x.extent(0);
+
+  // verify that x, y are different before running kernel
+  for (std::size_t i=0; i<extent; ++i){
+    EXPECT_TRUE(x(i)!=y(i));
+  }
+
+  // copy x and y before kernel
+  auto x_preKernel = kokkostesting::create_stdvector_and_copy(x);
+  auto y_preKernel = kokkostesting::create_stdvector_and_copy(y);
+
+  stdla::swap_elements(KokkosKernelsSTD::kokkos_exec<>(), x, y);
+
+  // after kernel, x should be unchanged, y should be equal to x
+  if constexpr(std::is_same_v<value_type, float>){
+    for (std::size_t i=0; i<extent; ++i){
+      EXPECT_FLOAT_EQ(x(i), y_preKernel[i]);
+      EXPECT_FLOAT_EQ(y(i), x_preKernel[i]);
+    }
+  }
+
+  if constexpr(std::is_same_v<value_type, double>){
```

Comment:
It's correct to use `value_type` here.  PR #162 (probably incorrectly) uses `element_type` instead.  (Please use `value_type` generally when you want to test the type with any `const` or `volatile` qualifications removed.)